lib/experimental/fonts-library/class-wp-font-family-utils.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * Fonts Family Utils class.
+ *
+ * This file contains utils fot Font Family class.
+ *
+ * @package    Gutenberg
+ * @subpackage Fonts Library
+ * @since      X.X.X
+ */
+
+/**
+ * A class of utilities for working with the Font Library.
+ */
+class WP_Font_Family_Utils {
+
+	/**
+	 * Generates a filename for a font face asset.
+	 *
+	 * Creates a filename for a font face asset using font family, style, weight and extension information.
+	 *
+	 * @param array  $font_face The font face array containing 'fontFamily', 'fontStyle', and 'fontWeight' attributes.
+	 * @param string $url The URL of the font face asset, used to derive the file extension.
+	 * @param int    $i Optional counter for appending to the filename, default is 1.
+	 * @return string The generated filename for the font face asset.
+	 */
+	static public function get_filename_from_font_face( $font_face, $url, $i = 1 ) {
+		$extension = pathinfo( $url, PATHINFO_EXTENSION );
+		$family    = sanitize_title( $font_face['fontFamily'] );
+		$style     = sanitize_title( $font_face['fontStyle'] );
+		$weight    = sanitize_title( $font_face['fontWeight'] );
+		$filename  = "{$family}_{$style}_{$weight}";
```

Comment:
A micro-optimization suggestion is sanitize after building the string, i.e. rather than sanitizing each individual piece of the string. In other words, sanitize the string one time instead of three times.

```suggestion
		$filename  = sanitize_title( "{$font_face['fontFamily']}_{$font_face['fontStyle']}_{$font_face['fontWeight']}" );
```